flink-runtime/src/main/java/org/apache/flink/runtime/jobmaster/JobMasterServiceLeadershipRunner.java

```diff
@@ -338,32 +356,24 @@ private void confirmLeadership(
         FutureUtils.assertNoException(
                 leaderAddressFuture.thenAccept(
                         address -> {
-                            synchronized (lock) {
-                                if (isValidLeader(leaderSessionId)) {
-                                    LOG.debug("Confirm leadership {}.", leaderSessionId);
-                                    leaderElection.confirmLeadership(leaderSessionId, address);
-                                } else {
-                                    LOG.trace(
-                                            "Ignore confirming leadership because the leader {} is no longer valid.",
-                                            leaderSessionId);
-                                }
-                            }
+                            runIfValidLeader(
+                                    leaderSessionId,
+                                    () -> {
+                                        LOG.debug("Confirm leadership {}.", leaderSessionId);
+                                        leaderElection.confirmLeadership(leaderSessionId, address);
+                                    },
+                                    "confirming leadership");
                         }));
     }
 
     private void forwardResultFuture(
             UUID leaderSessionId, CompletableFuture<JobManagerRunnerResult> resultFuture) {
         resultFuture.whenComplete(
                 (jobManagerRunnerResult, throwable) -> {
```

Comment:
```suggestion
                (jobManagerRunnerResult, throwable) -> 
```
And while squashing the commits you can also remove the obsolete bracket here.

Reply:
Fixed.